webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/batch/AsyncDeploymentActor.scala

```diff
@@ -67,6 +68,31 @@ final case class AutomaticStartDeployment(modId: ModificationId, actor:EventActo
 //actor: the actor who asked for the deployment
 final case class ManualStartDeployment(modId: ModificationId, actor:EventActor, reason:String) extends HashcodeCaching with StartDeploymentMessage
 
+
+sealed trait PolicyGenerationTrigger
+object PolicyGenerationTrigger {
+  final case object AllGeneration extends PolicyGenerationTrigger
+  final case object OnlyManualGeneration extends PolicyGenerationTrigger
+  final case object NoGeneration extends PolicyGenerationTrigger
+
+  def apply(value : String) : Box[PolicyGenerationTrigger] = {
+
+    value.toLowerCase() match {
+      case "all" | "allgeneration" => Full(AllGeneration)
+      case "none" | "nogeneration" => Full(NoGeneration)
```

Comment:
when we are in "none" mode, it's a bit surprising to have nothing done when you click on "regenerate all". Perhaps we should disable that button, or just add an explanation text below "status" in the drop down: "generation mode: none ; delay: 5 minutes" with a long explanation on mouse over.

Reply:
perhaps we can had that message only in 6.2 to minimize changes in 5.0/6.1 (in these versions, it will be documented only in settings API)

Reply:
I opened https://issues.rudder.io/issues/18896 in 6.2